client/rest-high-level/src/test/java/org/elasticsearch/client/RequestConvertersTests.java

```diff
@@ -1448,6 +1451,18 @@ public void testGetRepositories() {
         assertThat(expectedParams, equalTo(request.getParameters()));
     }
 
+    public void testPutRepository() throws IOException {
+        String repository = "repo";
+        PutRepositoryRequest putRepositoryRequest = new PutRepositoryRequest(repository);
+        putRepositoryRequest.type(FsRepository.TYPE);
+        putRepositoryRequest.settings("{\"location\": \".\"}", XContentType.JSON);
```

Comment:
would it be possible to randomize the values that the request holds like in other tests?

Reply:
I assumed that there is no problem setting values and checking that the output of the conversion from high-level request to low-level request is the expected one. We don't validate etc. I would do only what is straight-forward.